dashboard/app/models/census/census_your_school2017v4.rb

```diff
@@ -0,0 +1,67 @@
+# == Schema Information
+#
+# Table name: census_submissions
+#
+#  id                           :integer          not null, primary key
+#  type                         :string(255)      not null
+#  submitter_email_address      :string(255)
+#  submitter_name               :string(255)
+#  submitter_role               :string(255)
+#  school_year                  :integer          not null
+#  how_many_do_hoc              :string(255)
+#  how_many_after_school        :string(255)
+#  how_many_10_hours            :string(255)
+#  how_many_20_hours            :string(255)
+#  other_classes_under_20_hours :boolean
+#  topic_blocks                 :boolean
+#  topic_text                   :boolean
+#  topic_robots                 :boolean
+#  topic_internet               :boolean
+#  topic_security               :boolean
+#  topic_data                   :boolean
+#  topic_web_design             :boolean
+#  topic_game_design            :boolean
+#  topic_other                  :boolean
+#  topic_other_description      :string(255)
+#  topic_do_not_know            :boolean
+#  class_frequency              :string(255)
+#  tell_us_more                 :text(65535)
+#  pledged                      :boolean
+#  created_at                   :datetime         not null
+#  updated_at                   :datetime         not null
+#
+# Indexes
+#
+#  index_census_submissions_on_school_year_and_id  (school_year,id)
+#
+
+# This class represents census submissions coming from /yourschool after
+# we fixed the issue that sometimes the followup questions were not being
+# shown when they should. This was also the first version of the form to
+# post an explicit version number.
+#
+class Census::CensusYourSchool2017v4 < Census::CensusYourSchool2017v3
```

Comment:
Where is the version specified? How do we know which type to use when a form is submitted?

Reply:
I have a pending test to change the forms to submit directly to the census_submissions table rather than writing to FORMS. I intend to make the versioning more explicit when I touch things for those changes. I took the opportunity to add a version number field in the /yourschool form when I fixed the i18n values issues. That is really just a placeholder to make the migration from FORMS to CENSUS_SUBMISSIONS a little easier.